.github/workflows/_setup_aks_cluster.yml

```diff
@@ -0,0 +1,110 @@
+          echo "Setting AKS_OIDC_ISSUER to $AKS_OIDC_ISSUER"
+
+          export USER_ASSIGNED_CLIENT_ID="$(az identity show --resource-group $RESOURCE_GROUP --name $USER_ASSIGNED_IDENTITY_NAME --query 'clientId' -otsv)"
+          echo "Setting USER_ASSIGNED_CLIENT_ID to $USER_ASSIGNED_CLIENT_ID"
+
+          echo "AKS_OIDC_ISSUER=$AKS_OIDC_ISSUER" >> $GITHUB_OUTPUT
+          echo "USER_ASSIGNED_CLIENT_ID=$USER_ASSIGNED_CLIENT_ID" >> $GITHUB_OUTPUT
+        
+      - name: Create Kafka Namespace 
+        id: create-kafka-namespace
+        if: steps.prepare-workload-identity-env-vars.outcome == 'success'
+        run: |
+          result=$(kubectl get namespace kafka 2>&1 || true)
+          if [[ $result == *"not found"* ]]; then
+              echo "kafka namespace not found. Create kafka namespace..."
+              kubectl create namespace kafka
+          else
+              echo "kafka namespace already exists."
+          fi
+      
+      - name: Create Service Account
+        id: create-service-account
+        if: steps.create-kafka-namespace.outcome == 'success'
+        run: |
+          kubectl delete sa -n kafka workload-identity-sa 2>&1 || true
+          cat <<EOF | kubectl apply -f - 
+          apiVersion: v1 
+          kind: ServiceAccount 
+          metadata: 
+            annotations: 
+              azure.workload.identity/client-id: ${{ steps.prepare-workload-identity-env-vars.outputs.USER_ASSIGNED_CLIENT_ID }}
+            name: workload-identity-sa 
+            namespace: kafka 
+          EOF
+      
+      - name: Setup Workload Identity
+        id: setup-workload-identity
+        if: steps.create-service-account.outcome == 'success'
+        run: |
+          result=$(az identity federated-credential show --name myFedIdentity --identity-name $USER_ASSIGNED_IDENTITY_NAME --resource-group $RESOURCE_GROUP 2>&1 || true)
+          if [[ $result == *${{ steps.prepare-workload-identity-env-vars.outputs.AKS_OIDC_ISSUER }}* ]]; then
+              echo "Federated identity already exists"
+          else
+              echo "Federated identity not found. Creating... "
+              az identity federated-credential create --name myFedIdentity --identity-name $USER_ASSIGNED_IDENTITY_NAME --resource-group $RESOURCE_GROUP --issuer ${{ steps.prepare-workload-identity-env-vars.outputs.AKS_OIDC_ISSUER }} --subject system:serviceaccount:kafka:workload-identity-sa
+          fi
+          
+      - name: Setup Kafka Cluster 
+        id: apply-kafka-cr-files
+        if: steps.setup-workload-identity.outcome == 'success'
+        run: |
+          kubectl create -f 'https://strimzi.io/install/latest?namespace=kafka' -n kafka 2>&1
+          # Apply the `Kafka` Cluster CR file
+          kubectl apply -f https://strimzi.io/examples/latest/kafka/kafka-persistent-single.yaml -n kafka 2>&1
+  
+          echo "Sleep for 3 minutes and wait for Kafka cluster to be created and fully working..."
+          sleep 180
```

Comment:
I don't think we should do this because the strimzi cluster is designed to be resilient when kafka pods are partially ready and working. By waiting, we could enter into a very long wait while the cluster is operating and functional. Plus, this part of the code is already working very well, I don't think we should make it more complicated.